### PR TITLE
Delete storage policy ids when cluser is deleted

### PR DESCRIPTION
**Pull request by gnufied**
cc @jcpowermac

### CODE REVIEW

**Comment by jcpowermac** [comment]
/test e2e-vsphere

cc: @rvanderp3

File: pkg/destroy/vsphere/vsphere.go [hunk 1]
```diff
@@ -212,6 +249,11 @@ func (o *ClusterUninstaller) Run() error {
 		o.Logger.Debug("No managed Folder found")
 	}
 
+	err = deleteStoragePolicy(context.TODO(), o.Client, o.InfraID)
```
[the remark this answers is not in the document]
Comment (gnufied) [reply pkg/destroy/vsphere/vsphere.go:252]: you can't tag storage policies AFAIK. I spent bit of time looking into that and I could not find an option.

File: pkg/destroy/vsphere/vsphere.go [hunk 2]
```diff
@@ -212,6 +249,11 @@ func (o *ClusterUninstaller) Run() error {
...
+	if err != nil {
+		return errors.Errorf("error deleting storage policy: %v", err)
+	}
+
 	o.Logger.Debug("Delete tag")
 	tagLogger := o.Logger.WithField("Tag", o.InfraID)
 	if err = deleteTag(context.TODO(), o.RestClient, o.InfraID); err != nil {
```
[the remark this answers is not in the document]
Comment (gnufied) [reply pkg/destroy/vsphere/vsphere.go:259]: Yes, deleting the tag removes the tag from associated objects (including datastore).

File: pkg/destroy/vsphere/vsphere.go [hunk 3]
```diff
@@ -130,6 +133,40 @@ func getAttachedObjectsOnTag(ctx context.Context, client *rest.Client, tagName s
...
+	policyName := fmt.Sprintf("openshift-storage-policy-%s", infraID)
+
+	matchingProfileIds := []pbmtypes.PbmProfileId{}
+	for _, p := range profiles {
+		if p.GetPbmProfile().Name == policyName {
+			profileID := p.GetPbmProfile().ProfileId
+			matchingProfileIds = append(matchingProfileIds, profileID)
+		}
+	}
+	_, err = pbmClient.DeleteProfile(ctx, matchingProfileIds)
```
[the remark this answers is not in the document]
Comment (gnufied) [reply pkg/destroy/vsphere/vsphere.go]: added logging

**Pull request merged**